pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,216 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	v1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	listerv1 "github.com/openshift/client-go/config/listers/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	controlplanev1 "github.com/openshift/api/kubecontrolplane/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+const (
+	operatorNamespace = operatorclient.OperatorNamespace
+	targetNamespace   = operatorclient.TargetNamespace
+
+	ApiServerConfigMapName = "config"
+	ApiServerConfigMapKey  = "config.yaml"
+
+	DefaultNotReadyTolerationSecondsArgument    = "default-not-ready-toleration-seconds"
+	DefaultUnreachableTolerationSecondsArgument = "default-unreachable-toleration-seconds"
+)
+
+// LatencyProfileController
+type LatencyProfileController struct {
```

Comment:
it would be nice if you could document the purpose of this controller

Reply:
Added a summary.
```

// LatencyProfileController periodically lists the config/v1/node object
// and fetches the worker latency profile applied on the cluster which is used to
// updates the status of the config node object. The status updates reflect the
// state of kube-api-server(s) running on control plane node(s) and their
// observed config for default-not-ready-toleration-seconds and
// default-unreachable-toleration-seconds match the applied arguments.
```